### ISSUE TITLE
Cannot disable logging

### ISSUE THREAD

**Issue by kbuilds** [AuthorizeNet/sdk-python#86]
Looks like the logging info is writing to a file called `anetsdk.log` by default. I need to disable this log, as all messages are being handled by exceptions, and the log file is causing issues with permissions on every server that my software is being deployed to.

Any ideas on how to disable this via configuration?

**Comment by rewindcaz** [comment]
Also wondering about this

**Comment by rewindcaz** [comment]
At the very least, is it possible to over-ride variables inside constant.py? Then setting a variable like defaultLoggingLevel to None might disable it?

**Comment by kbuilds** [comment]
@brianmc Are you the owner of this project at Auth.net?

**Comment by rneilson** [comment]
Honestly, I do a dirty, filthy monkey patch of the `apicontrollers` module:

```python
import logging
from authorizenet import apicontractsv1, apicontrollers

log = logging.getLogger('creditcard')

if not settings.AUTHORIZE_NET_LOG:
    log.setLevel(logging.CRITICAL)

def _fake_config (*args, **kwargs): # pragma: no cover
    pass
log.basicConfig = _fake_config
apicontrollers.logging = log
apicontrollers.apicontrollersbase.logging = log
```

`apicontrollersbase` sets up a logger, and calls `.basicConfig()` on it to set its level and output file, so I monkey-patch that with an empty function, and set my real logging config beforehand. (The `settings` listed above happens to be a Django settings object, so pick your conditional to taste.)

Side note: by importing `apicontrollers` as a module instead of importing `*` from it, that A) prevents `logging` from getting clobbered in the namespace, and B) allows only monkey-patching `apicontrollers.logging`, leaving you free to log what you actually want to.

**Comment by kbuilds** [comment]
@rneilson I feel like I need to shower after reading this. But if it works it works.

On a serious note, where are the maintainers for this repo?

**Comment by rneilson** [comment]
@kbuilds I almost went to confession after writing it, and I'm not even Catholic.

It mostly works -- there are still some stray error messages from `pyxb`, and sometimes it'll output a `ContentNondeterminismError` instead of stringified XML on some calls, so it's not as useful as I'd like for the times I _do_ want to log requests and responses. But it'll definitely prevent that log file from being created.

As for maintainers: @Sapbasu15? @sunnyrajrathod?

**Comment by Sapbasu15** [comment]
@rneilson @kbuilds we are planning to move default logging to stderr stream with critical level (almost similar to no logging), and logging level and destination (file/stream) will be configurable via ini file. Also you'll be able to add handlers to root logger and configure it if required.
Will that solve the problems you are facing regarding logging?

**Comment by rneilson** [comment]
@Sapbasu15 I would actually vastly prefer _no_ logger configuration be done inside the library, just a standard named logger I can configure as I wish, no extra .ini file required. (If you do set any defaults on the logger, please make sure they're done at the module's top level, not when instantiating a controller like it is now.)

**Comment by Sapbasu15** [comment]
@rneilson we can provide a named logger like 'authorizenet_sdk' and add a null handler to it, assuming a configured handler to be added by developer. Currently its using the root logger only. It seems root logger has to be configured to avoid 'no logger' messages from pyxb.  Also we think it'll be good to provide some defaults for ease of use, which can be done at module level rather than base constructor.

**Comment by rneilson** [comment]
@Sapbasu15 Quite honestly, if the library configures anything on the root logger, that would be just as bad as the current setup, because it's likely to clobber (or, almost as bad, silently change) existing configuration. And I suspect if you set a null handler for the `pyxb` named logger as well (at the module level), that'll get rid of the `No handlers could be found for logger "pyxb.binding.content"` messages too.

Just don't touch my root logger, y'know?

**Comment by Sapbasu15** [comment]
Fix for the issues is released in PR #102 and has been released in the latest version.
